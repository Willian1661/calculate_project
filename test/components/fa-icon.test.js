const numbersIcon = require("../../src/components/fa-icons/numbers-icon")
const asterisk = require("../../src/components/fa-icons/asterisk-icon")
const clearAll = require("../../src/components/fa-icons/clear-all-icon")
const delet = require("../../src/components/fa-icons/delete-icon")
const devide = require("../../src/components/fa-icons/divide-icon")
const equals = require("../../src/components/fa-icons/equals-icon")
const minus = require("../../src/components/fa-icons/minus-icon")
const percent = require("../../src/components/fa-icons/percent-icon")
const period = require("../../src/components/fa-icons/period-icon")
const plus = require("../../src/components/fa-icons/plus-icon")
const plusMinus = require("../../src/components/fa-icons/plus-minus-icon")

describe("Font Awesome icons", () =>{
    const operators = [asterisk(),devide(),plus(),plusMinus(),minus(),percent(),]
    const especials = [clearAll(),delet(),equals(),period()]
    const numbers = ["1","2","3","4","5","6","7","8","9","0",]

    it.each(numbers)("'s passing fa-icons correctly", (number) => {
        expect(numbersIcon(number)).toMatchSnapshot()
    });

    it("'s passing especial icons correctly", () => {
        expect(especials).toMatchSnapshot()
    });

    it("'s passing operators icon correctly", () => {
        expect(operators).toMatchSnapshot()
    });
    
})