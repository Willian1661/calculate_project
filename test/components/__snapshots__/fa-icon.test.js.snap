// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Font Awesome icons 's passing especial icons correctly 1`] = `
[
  "<i class="fa-solid fa-c" data-value="C"></i>",
  "<i class="fa-solid fa-delete-left" data-value="D"></i>",
  "<i class="fa-solid fa-equals" data-value="="></i>",
  "<i class="fa-solid fa-period" data-value=".">.</i>",
]
`;

exports[`Font Awesome icons 's passing fa-icons correctly 1`] = `"<i class="fa-solid fa-1" data-value="1"></i>"`;

exports[`Font Awesome icons 's passing fa-icons correctly 2`] = `"<i class="fa-solid fa-2" data-value="2"></i>"`;

exports[`Font Awesome icons 's passing fa-icons correctly 3`] = `"<i class="fa-solid fa-3" data-value="3"></i>"`;

exports[`Font Awesome icons 's passing fa-icons correctly 4`] = `"<i class="fa-solid fa-4" data-value="4"></i>"`;

exports[`Font Awesome icons 's passing fa-icons correctly 5`] = `"<i class="fa-solid fa-5" data-value="5"></i>"`;

exports[`Font Awesome icons 's passing fa-icons correctly 6`] = `"<i class="fa-solid fa-6" data-value="6"></i>"`;

exports[`Font Awesome icons 's passing fa-icons correctly 7`] = `"<i class="fa-solid fa-7" data-value="7"></i>"`;

exports[`Font Awesome icons 's passing fa-icons correctly 8`] = `"<i class="fa-solid fa-8" data-value="8"></i>"`;

exports[`Font Awesome icons 's passing fa-icons correctly 9`] = `"<i class="fa-solid fa-9" data-value="9"></i>"`;

exports[`Font Awesome icons 's passing fa-icons correctly 10`] = `"<i class="fa-solid fa-0" data-value="0"></i>"`;

exports[`Font Awesome icons 's passing operators icon correctly 1`] = `
[
  "<i class="fa-solid fa-asterisk" data-value="*"></i>",
  "<i class="fa-solid fa-divide" data-value="/"></i>",
  "<i class="fa-solid fa-plus" data-value="+"></i>",
  "<i class="fa-solid fa-plus-minus" data-value="+/-"></i>",
  "<i class="fa-solid fa-minus" data-value="-"></i>",
  "<i class="fa-solid fa-percent" data-value="%"></i>",
]
`;
